File: extract-pdf.js
import fs from 'fs';
import pdfParse from 'pdf-parse/lib/pdf-parse.js';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const pdfDir = path.join(__dirname, 'public', 'PDF');
const pdfFiles = fs.readdirSync(pdfDir).filter(f => f.endsWith('.pdf'));

async function extractPDFContent() {
  for (const file of pdfFiles) {
    const filePath = path.join(pdfDir, file);
    const dataBuffer = fs.readFileSync(filePath);
    
    try {
      const data = await pdfParse(dataBuffer);
      console.log('\n========================================');
      console.log(`📄 ARCHIVO: ${file}`);
      console.log('========================================');
      console.log(data.text);
      console.log('\n');
    } catch (error) {
      console.error(`Error leyendo ${file}:`, error.message);
    }
  }
}

extractPDFContent();

File: extract-pdf.py
import PyPDF2
import os
from pathlib import Path

pdf_dir = Path(r'C:\Projects\Edit-Assistant\public\PDF')
pdf_files = sorted(pdf_dir.glob('*.pdf'))

for pdf_file in pdf_files:
    print('\n' + '='*60)
    print(f'📄 ARCHIVO: {pdf_file.name}')
    print('='*60)
    
    try:
        with open(pdf_file, 'rb') as file:
            pdf_reader = PyPDF2.PdfReader(file)
            text = ''
            for page in pdf_reader.pages:
                text += page.extract_text()
            print(text)
    except Exception as e:
        print(f'Error: {e}')
    
    print('\n')
